internal/handlers/users_v1_handler.go

```diff
@@ -1,18 +1,86 @@
 package handlers
 
 import (
+	"context"
+	"encoding/json"
+	"io"
 	"net/http"
 
 	"github.com/redhatinsights/mbop/internal/config"
+	"github.com/redhatinsights/mbop/internal/service/ocm"
+
+	"github.com/redhatinsights/mbop/internal/models"
 )
 
 func UsersV1Handler(w http.ResponseWriter, r *http.Request) {
+	ctx := context.Background()
+
 	switch config.Get().UsersModule {
 	case awsModule:
-		// acctNum := chi.URLParam(r, "accountNumber")
-		// users, err := users.ListUsersV1(acctNum)
-		// ...
-		// w.Write(users.ToJSON())
+		body, err := io.ReadAll(r.Body)
+		if err != nil {
+			do500(w, "failed to read request body: "+err.Error())
+			return
+		}
+		defer r.Body.Close()
+
+		var usernames models.UserBody
+		err = json.Unmarshal(body, &usernames)
+		if err != nil {
+			do400(w, "failed to parse request body: "+err.Error())
```

Comment:
I wonder if it's worth having a message around requiring a POST payload with a collection of usernames `"usernames": []`? Testing locally it felt like the error `failed to parse request body: unexpected end of JSON input` was more an app error, when actually it was me not passing expected data.

Reply:
gotcha!  I'll add that in!  I think that makes more sense!